main.go

```diff
@@ -21,67 +22,101 @@ func failf(msg string, args ...interface{}) {
 	os.Exit(1)
 }
 
+func isReleaseChannel(versionString string) bool {
+	releaseChannels := [...]string{"stable", "beta", "dev", "master"}
+	for _, channel := range releaseChannels {
+		if channel == versionString {
+			return true
+		}
+	}
+	return false
+}
+
 func main() {
 	var cfg config
 	if err := stepconf.Parse(&cfg); err != nil {
 		failf("Issue with input: %s", err)
 	}
 	stepconf.Print(cfg)
 
-	fmt.Println()
-	log.Infof("Downloading Flutter SDK")
-	log.Printf("git clone")
-
-	sdkLocation := filepath.Join(os.Getenv("HOME"), "flutter-sdk")
-
-	if err := os.RemoveAll(sdkLocation); err != nil {
-		failf("Failed to remove path(%s), error: %s", sdkLocation, err)
-	}
-
-	gitRepo, err := git.New(sdkLocation)
+	preInstalled := true
+	_, err := exec.LookPath("flutter")
 	if err != nil {
-		failf("Failed to open git repo, error: %s", err)
-	}
-
-	if err := gitRepo.CloneTagOrBranch("https://github.com/flutter/flutter.git", cfg.Version).Run(); err != nil {
-		failf("Failed to clone git repo for tag/branch: %s, error: %s", cfg.Version, err)
+		preInstalled = false
+		log.TPrintf("Flutter is not preinstalled.")
+	} else {
+		log.TInfof("Preinstalled Flutter version:")
+		versionCmd := command.New("flutter", "--version").SetStdout(os.Stdout).SetStderr(os.Stderr)
+		log.Donef("$ %s", versionCmd.PrintableCommandArgs())
+		fmt.Println()
+		if err := versionCmd.Run(); err != nil {
+			failf("Failed to check flutter version, error: %s", err)
+		}
 	}
 
-	log.Printf("set in $PATH")
-
-	path := filepath.Join(sdkLocation, "bin") + ":" + os.Getenv("PATH")
-
-	if err := os.Setenv("PATH", path); err != nil {
-		failf("Failed to set env, error: %s", err)
+	if preInstalled && isReleaseChannel(cfg.Version) {
+		fmt.Println()
+		log.TInfof("Setting flutter channel")
+		channelCmd := command.New("flutter", "channel", cfg.Version).SetStdout(os.Stdout).SetStderr(os.Stderr)
+		log.Donef("$ %s", channelCmd.PrintableCommandArgs())
+		fmt.Println()
+		if err := channelCmd.Run(); err != nil {
+			failf("Failed to set flutter channel, error: %s", err)
+		}
+
+		fmt.Println()
+		log.TInfof("Upgrading flutter")
+		upgradeCmd := command.New("flutter", "upgrade").SetStdout(os.Stdout).SetStderr(os.Stderr)
+		log.Donef("$ %s", channelCmd.PrintableCommandArgs())
+		fmt.Println()
+		if err := upgradeCmd.Run(); err != nil {
+			failf("Failed to set flutter channel, error: %s", err)
+		}
+	} else {
+		fmt.Println()
+		log.TInfof("Downloading Flutter SDK")
+		log.Printf("git clone")
+
+		sdkLocation := filepath.Join(os.Getenv("HOME"), "flutter-sdk")
+
+		if err := os.RemoveAll(sdkLocation); err != nil {
```

Comment:
I think it would be a good idea to add at least a debug log about the `os.RemoveAll(sdkLocation)`.
'Cause currently, the step says that it will download the Flutter via git clone, and instead of that the code will remove the old files instead. 
This could mislead the user.